### PR TITLE
(GH-2892) Document sensitive task output limitations

### LINKED ISSUE #2892: Sensitive task output not supported for pcp transport
## Describe the Bug
Plans that attempt to use the `Result` from tasks run over PCP with `_sensitive` output are unable to `unwrap` the result.

## Expected Behavior
Sensitive task results usable run over PCP

## Additional Context
The orchestrator `task` command will always return a `Sensitive: [value redacted]` string (as it is not stored in the orch database). 

## Solution

Document this behavior in "Writing/Running Tasks". Potentially reference it from "Using Bolt with Puppet Enterprise".

### PR DESCRIPTION
**Pull request by beechtom**
This adds a section to the 'Troubleshooting' page that documents
limitations around running tasks with sensitive output using the `pcp`
transport.

!no-release-note

Closes #2892

### CODE REVIEW

File: documentation/troubleshooting.md
Comment (artlawson) [inline documentation/troubleshooting.md:378]: I feel silly for asking this, but are you writing the `sensitive_task_output.md` because I tried the link and it didn't work?

**Review by artlawson — COMMENTED** [review]
otherwise looks good so far

File: documentation/troubleshooting.md
Comment (beechtom) [reply documentation/troubleshooting.md:378]: @artlawson That file hasn't been merged in yet. Once it is, the link will work. :) Relevant PR is [here](https://github.com/puppetlabs/puppetlabs-bolt_shim/pull/25).

Comment (artlawson) [reply documentation/troubleshooting.md:378]: 😅😅

**Review by hestonhoffman — APPROVED** [review]
This is good to merge once the dependant doc is merged in :)

**Pull request merged**
